Can not get proxyquire to work with a require that happens in an init function

I have some code like this:

/lib/mine.js
```
const path = require('path');

function init(cfg) {
  var inner = require(cfg.file);
  inner.doSomething();
}

module.exports = {init};
```

/app.js
```
const mine = require(('./lib/mine');
// Do some stuff here

const cfg = {
  file: '../otherPath/otherFile'
};

mine.init(cfg);
```

The code is much more complicated then this, but this is close enough to show the problem.

I am writing tests for `app.js` above and want to use `proxyquire` for the file `../otherPath/otherFile`

My test file is like this:

app.mocha.js
```
const proxyquire = require('proxyquire').noCallThru();

const ofStub = {
  doSomething: function() {
    // Do something
  }
}

const app = proxyquire('./app', {
  '../otherPath/otherFile': ofStub
});

```

The rest of this file does not matter since the error happens when I try to use `proxyquire` on `app`

I get something like the following error:

`Error: Cannot find module '../otherPath/otherFile`

The thing is that it should not have tried to load that module yet since I have not called `init`

Is there a way to force `proxyquire` to not care if the real module was loaded using `require` when `proxyquire` is initially called and just allow the module to be overridden if and when it is loaded?

--- a/lib/proxyquire.js
+++ b/lib/proxyquire.js
@@ -154,7 +154,7 @@ Proxyquire.prototype._resolveModule = function (baseModule, pathToResolve) {
     // directory.  However, if !this._preserveCache, then we don't want to
     // throw, since we can resolve modules that don't exist.  Resolve as
     // best we can.
-    if (!this._preserveCache) {
+    if (!this._preserveCache || this._noCallThru) {
       return path.resolve(path.dirname(baseModule), pathToResolve)
     }
 

--- a/test/proxyquire-notexisting.js
+++ b/test/proxyquire-notexisting.js
@@ -15,13 +15,22 @@ describe('When resolving foo that requires stubbed /not/existing/bar.json', func
   })
 })
 
-describe('When resolving foo that requires stubbed /not/existing/bar.json with noCallThru', function () {
-  var foo
-
+describe('When resolving foo that requires stubbed /not/existing/bar.json with @noCallThru', function () {
   it('resolves foo with stubbed bar', function () {
-    foo = proxyquire(fooPath, {
+    var foo = proxyquire(fooPath, {
       '/not/existing/bar.json': { config: 'bar\'s config', '@noCallThru': true }
     })
     assert.equal(foo.config, 'bar\'s config')
+  })
+})
+
+describe('When resolving foo that requires stubbed /not/existing/bar.json with noCallThru()', function () {
+  it('resolves foo with stubbed bar', function () {
+    proxyquire.noCallThru()
+    var foo = proxyquire(fooPath, {
+      '/not/existing/bar.json': { config: 'bar\'s config' }
+    })
+    assert.equal(foo.config, 'bar\'s config')
+    proxyquire.callThru()
   })
 })

